src/sync/app.cpp

```diff
@@ -561,19 +563,13 @@ void App::get_profile(std::function<void(std::shared_ptr<SyncUser>, Optional<App
     };
     
     std::string profile_route = util::format("%1/auth/profile", m_base_route);
-    std::string bearer = util::format("Bearer %1", sync_user->access_token());
-
-    m_config.transport_generator()->send_request_to_server({
-        HttpMethod::get,
-        profile_route,
-        m_request_timeout_ms,
-        {
-            { "Content-Type", "application/json;charset=utf-8" },
-            { "Accept", "application/json" },
-            { "Authorization", bearer }
-        },
-        std::string(),
-        }, profile_handler);
+    
+    do_authenticated_request({
```

Comment:
Shouldn't all REST methods use this method? I assume we want to refresh the token no matter which API called failed because of it?

Reply:
calls which require authentication now use `do_authenticated_request`